Arduino/RobotLightsSPI/RobotLightsSPI.ino

```diff
@@ -0,0 +1,244 @@
+   - 0x04 : start address ; RGB ; length ; stride : set "length" pixels starting at "start" with value RGB.  Repeat every "stride" pixels.
+   - 0x05 : start address ; RGB ; length ; stride ; max length : same as 0x04, but will stop after max length pixels
+
+   Error rejection/correction:
+   - Invalid packets will be rejected and packet parsing will begin again with next byte in stream.
+
+   Display protocol:
+   - At beginning of frame, send >= 16 0xff bytes to force resync
+   - Send paint commands to fill image buffer.  Unpainted pixels will retain prior contents.
+   - Send "show current image" command
+
+   NeoPixels will hold the last value they received.  There is no required refresh rate.
+
+*/
+
+// Max pixels on Trinket is 94 due to RAM limits
+
+#if defined(ARDUINO_AVR_UNO)
+#define PIN 1
+#define MAXLEDS 240
+
+#elif defined(ARDUINO_AVR_PRO)
+#define PIN 3
+#define MAXLEDS 240
+
+#else
+#error unsupported board.  Please add to code.
+#endif
+
+const uint32_t dimRED = 0x200000;
+const uint32_t dimGREEN = 0x002000;
+const uint32_t dimBLUE = 0x000020;
+const uint32_t dimWHITE = 0x202020;
+
+// Parameter 1 = number of pixels in strip
+// Parameter 2 = Arduino pin number (most are valid)
+// Parameter 3 = pixel type flags, add together as needed:
+//   NEO_KHZ800  800 KHz bitstream (most NeoPixel products w/WS2812 LEDs)
+//   NEO_KHZ400  400 KHz (classic 'v1' (not v2) FLORA pixels, WS2811 drivers)
+//   NEO_GRB     Pixels are wired for GRB bitstream (most NeoPixel products)
+//   NEO_RGB     Pixels are wired for RGB bitstream (v1 FLORA pixels, not v2)
+//   NEO_RGBW    Pixels are wired for RGBW bitstream (NeoPixel RGBW products)
+
+Adafruit_NeoPixel strip = Adafruit_NeoPixel(MAXLEDS, PIN, NEO_GRB + NEO_KHZ800);
+
+// IMPORTANT: To reduce NeoPixel burnout risk, add 1000 uF capacitor across
+// pixel power leads, add 300 - 500 Ohm resistor on first pixel's data input
+// and minimize distance between Arduino and first pixel.  Avoid connecting
+// on a live circuit...if you must, connect GND first.
+
+const int MAX_PAYLOAD = 15;
+uint8_t payload[MAX_PAYLOAD];
+uint32_t lastPacket;
+
+const int RECV_BUFF_SIZE = MAX_PAYLOAD * 4;
+uint8_t recvBuff[RECV_BUFF_SIZE];
+volatile int recvQueueHead = 0;
+volatile int recvQueueTail = 0;
+
+// The SPI library only supports configuring the arduino as an SPI controller,
+// so we have to manually interface with the SPI hardware
+// See: https://arduino.stackexchange.com/questions/76528/spi-slave-programming-for-arduino
+ISR(SPI_STC_vect) {
+  volatile int nextQueueTail = (recvQueueTail + 1) % RECV_BUFF_SIZE;
+
+  // If the receive buffer has filled, ignore all incoming data until
+  // space is available.
+  if(nextQueueTail == recvQueueHead) {
+    return;
+  }
+
+  recvBuff[recvQueueTail] = SPDR;
+  recvQueueTail = nextQueueTail;
+}
+
+void setup() {
+  lastPacket = millis();
+
+  strip.begin();
+
+  pinMode(MOSI, INPUT);
```

Comment:
A fair bit. However, SPI uses dedicated hardware, so you can't adjust the pins in software unless you use software SPI, which has speed/throughput limitations.

On the arduino side, we need to make three changes:
1. Move the data wire from pin A4 (SDA) to pin 11 (MOSI)
2. Move the clock from pin A5 (SCL) to pin 13 (SCK)
3. Add a pulldown resistor (~10KΩ) between pin 10 (SS) and ground

Since we never send data from the LEDs to the rio, I don't think we need to bother connecting MISO. 

On the rio side, we basically need to cut off the i2c adapter, and solder MOSI to the data wire, SCL to the clock wire, and ground to ground. I've already bought a few connectors on amazon (<https://www.amazon.com/dp/B07SR3S2W3>), so  assuming I bought the right connectors, that should be pretty simple